wdl-parsing/WDLParsingFunction/src/main/java/io/dockstore/wdlparser/App.java

```diff
@@ -0,0 +1,196 @@
+import com.amazonaws.services.lambda.runtime.RequestHandler;
+import com.amazonaws.services.lambda.runtime.events.APIGatewayProxyRequestEvent;
+import com.amazonaws.services.lambda.runtime.events.APIGatewayProxyResponseEvent;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import dockstore.openapi.client.model.LanguageParsingRequest;
+import dockstore.openapi.client.model.LanguageParsingResponse;
+import dockstore.openapi.client.model.VersionTypeValidation;
+import java.io.File;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import java.net.HttpURLConnection;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import javax.ws.rs.core.MediaType;
+import org.eclipse.jgit.api.Git;
+import org.eclipse.jgit.api.errors.GitAPIException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import scala.Option;
+import scala.collection.JavaConverters;
+import womtool.WomtoolMain;
+
+/** Handler for requests to Lambda function. */
+public class App
+    implements RequestHandler<APIGatewayProxyRequestEvent, APIGatewayProxyResponseEvent> {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(App.class);
+  ObjectMapper mapper = new ObjectMapper();
+
+  /**
+   * Get a language parsing response by running womtool.
+   *
+   * @param descriptorAbsolutePathString Absolute path to the main descriptor file
+   * @return LanguageParsingResponse constructed after running womtool
+   */
+  public static LanguageParsingResponse getResponse(String descriptorAbsolutePathString) {
+    LanguageParsingResponse response = new LanguageParsingResponse();
+    response.setClonedRepositoryAbsolutePath(descriptorAbsolutePathString);
+    List<String> commandLineArgs = Arrays.asList("validate", "-l", descriptorAbsolutePathString);
+    try {
+      // TODO: This can be cleaner (without parsing stdout) by directly using Validate.scala, etc
+      WomtoolMain.Termination termination =
+          WomtoolMain.runWomtool(
+              JavaConverters.collectionAsScalaIterableConverter(commandLineArgs).asScala().toSeq());
+      Option<String> stdout = termination.stdout();
+      // The womtool successful response returns lines of text.
+      String[] split = stdout.get().split(System.lineSeparator());
+      List<String> strings = new ArrayList<>();
+      Collections.addAll(strings, split);
+
+      // The first two lines aren't actual paths.
+      VersionTypeValidation versionTypeValidation = new VersionTypeValidation();
+      if (strings.get(0).equals("Success!")
+          && strings.get(1).equals("List of Workflow dependencies is:")) {
+        versionTypeValidation.setValid(true);
+        response.setVersionTypeValidation(versionTypeValidation);
+        handleSuccessResponse(response, strings);
```

Comment:
`strings` can include fully qualified http(s) imports. Is it your intent that they be part of the response? I can see it either way.

```
java -jar ~/Downloads/womtool-54.jar  validate -l Dockstore.wdl
Success!
List of Workflow dependencies is:
https://raw.githubusercontent.com/coverbeck/bravo-pipeline/noroot/vcfPercentilesPreparation.wdl
https://raw.githubusercontent.com/coverbeck/bravo-pipeline/noroot/cramPreparation.wdl
https://raw.githubusercontent.com/coverbeck/bravo-pipeline/noroot/coveragePreparation.wdl
```

Reply:
yes